tests/e2e/examples.test.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright 2024 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { execSync } from "child_process";
+import * as fs from "fs";
+import * as path from "path";
+
+const pattern = process.env.EXAMPLE_NAME;
+const files: string[] = [];
+function getTypeScriptFiles(directoryPath: string) {
+  const fileNames = fs.readdirSync(directoryPath);
+
+  for (const fileName of fileNames) {
+    const filePath = path.join(directoryPath, fileName);
+    if (fs.statSync(filePath).isDirectory()) {
+      getTypeScriptFiles(filePath);
+    } else {
+      if (filePath.endsWith(".ts")) {
+        files.push(filePath);
+      }
+    }
+  }
+}
+
+function runExample(example: string) {
+  describe("e2e test", () => {
+    it(`Runs ${example}`, async () => {
+      execSync(`yarn start -- ${example}`);
+    });
+  });
+}
+
+const exampleDirectory = "./examples";
+getTypeScriptFiles(exampleDirectory);
+
+for (const example of files) {
+  if (pattern == null || pattern.trim() == "") {
+    runExample(example);
+  } else {
+    if (example.includes(pattern)) {
+      runExample(example);
+    }
+  }
+}
```

Comment:
The script could be clearer and easier to follow. However, you got the point.

Let's just simply it to something like this?

```bash
# glob is already installed (just types are missing)
yarn add -D @types/glob
```

```typescript
import { exec } from "child_process";
import { promisify } from "util";
import { glob } from "glob";

const execAsync = promisify(exec);
const pattern = process.env.EXAMPLE_NAME || `./examples/**/*.ts`;

describe("e2e test", async () => {
  const exampleFiles = await glob(pattern, {
    cwd: process.cwd(),
    dot: false,
    realpath: true,
  });

  it.each(exampleFiles)("Runs %s", async (example) => {
    const { stderr, stdout } = await execAsync(`yarn start -- ${example}`);
    console.log({
      path: example,
      stdout,
    });
    expect(stderr).toBeFalsy();
  });
});
```


Also, please try to play with the Vitest concurrency. We will not typically want to run all examples in a sequence but rather in parallel (with some limit).

Reply:
Thanks for the updates.  It is much simpler now.  I changed to run the examples in parallel.  I haven't add the `some limit` yet.

Reply:
Yes. However, in that case, I would not call these `E2E Tests` but rather `Examples Tests` or something like that.